libraries/chain/database.cpp

```diff
@@ -3404,15 +3405,58 @@ void database::clear_expired_delegations()
       itr = delegations_by_exp.begin();
    }
 }
+#ifdef STEEM_ENABLE_SMT
+template< typename smt_balance_object_type >
+void database::adjust_smt_balance( const account_object& a, const asset& delta, bool check_balance )
```

Comment:
Only the account name is used in this method, so let's just pass the account name.